static/app/views/dashboardsV2/manage/index.tsx

```diff
@@ -45,9 +48,23 @@ type Props = {
 type State = {
   dashboards: DashboardListItem[] | null;
   dashboardsPageLinks: string;
+  showTemplates: boolean;
 } & AsyncView['state'];
 
 class ManageDashboards extends AsyncView<Props, State> {
+  state: State = {
+    // AsyncComponent state
+    loading: true,
+    reloading: false,
+    error: false,
+    errors: {},
+
+    // local component state
+    dashboards: null,
+    dashboardsPageLinks: '',
+    showTemplates: shouldShowTemplates(),
+  };
+
```

Comment:
I think you can do something like this so you don't have to redefine the defaults

```suggestion
  getDefaultState() {
    return {
      ...super.getDefaultState(),
      showTemplates: shouldShowTemplates(),
    };
  }

```